Automattic/jetpack
# Jetpack Sync: Add Sync lock related info in Sync debug details

To help with debugging Sync related issues we are adding additional info on Sync locks in Sync debug details.

#### Changes proposed in this Pull Request:

* `Sync/Actions::get_debug_details`: Adds the following Sync lock related info in the debug details grouped as `sync_locks`:

```
$debug['debug_details']['sync_locks'] = array(
	'retry_time_sync'  // Retry time lock for Sync queue related to WPCOM sending us a `Retry-After` header
	'retry_time_full_sync'  // Retry time lock for Full Sync queue related to WPCOM sending us a `Retry-After` header
	'next_sync_time_sync'   // Next Sync lock for Sync queue, usually set on error responses when attempting to send Sync actions
	'next_sync_time_full_sync' // Next Sync lock for Full Sync queue, usually set on error responses when attempting to send Sync actions
	'queue_locked_sync'  // If the Sync queue is locked.
	'queue_locked_full_sync'  // If the Full Sync queue is locked.
	'dedicated_sync_request_lock' // The lock that Dedicated Sync sets to prevent spawning concurrent Sync requests
)
```

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
Try fetching the Sync status with debugging enabled on your WPCOM sandbox and confirm the additional debug data are present.
